flask/app.py

```diff
@@ -825,14 +825,13 @@ def run(self, host=None, port=None, debug=None, **options):
                         information.
         """
         from werkzeug.serving import run_simple
-        if host is None:
-            host = '127.0.0.1'
-        if port is None:
-            server_name = self.config['SERVER_NAME']
-            if server_name and ':' in server_name:
-                port = int(server_name.rsplit(':', 1)[1])
-            else:
-                port = 5000
+
+        _host = '127.0.0.1'
+        _port = 5000
+        server_name = self.config.get("SERVER_NAME", None)
+        if server_name:
+            _host, _port = servername.split(':', 1)
+        host, port = host or _host, port or int(_port)
```

Comment:
Good catch. I forgot about that. I like the way this is structured though, reads much cleaner to my eyes than the previous logic.